content/specific-skill-success-criteria/classes-and-objects/_index.md

```diff
@@ -23,3 +23,122 @@ All students need to provably understand all of the following concepts:
 - inheritance and overriding and extending methods
 - composition
+
+# Python
+
+## Constructors
+The `__init__()` method is called the constructor and is always called when an object is created, also known as instantiating an object.
+
+### Default Constructor
+The default constructor is a simple constructor which doesn’t accept any arguments. Its definition has only one argument which is a reference to the instance being constructed.
+```
+class BankAccount:
+  def __init__(self):
+    pass
+```
+
+### Parameterized Constructor
+As the name suggests, a parameterized constructor is a constructor with parameters. Takes its first argument as a reference to the instance being constructed, known as `self` and the rest of the arguments are provided by the programmer.
+```
+class BankAccount:
+  def __init__(self, acc_no, amount):
+    self.acc_no = acc_no
+    self.amount = amount
+```
+
+## Instantiation
+Instantiating a class is creating a copy of the class which inherits all class variables and methods. We simply call the class as if it were a function, passing the arguments that the `__init__()` method defines (except `self`).
+```
+saitama_acc = BankAccount("sa1234", 10000)
+```
+
+A class can have multiple instances.
+```
+kuroko_acc = BankAccount("ka1234", 300)
+tsukasa_acc = BankAccount("ta1234", 75010)
+vash_acc = BankAccount("va1234", 813650)
+```
+
+
+## Getters and setters
+The main purpose that getters and setters serve in object-orientated programs is for ensuring data encapsulation - or put another way, as a mechanism for hiding the implementation details of a class from the user. In Python, getters and setters are not the same as for other object-orientated languages. Private variables in Python are not actually hidden fields. In Python getters and setters are typically used when:
+
+- we want to add validation logic for getting and setting a value
+- to avoid the direct access of a class field so that private variables cannot be accessed directly/modified by an outside user
+
+### Getters
+A getter is a method that gets the value of a property.
+```
+  def balance(self):
+    return self.amount
+```
+
+```
+saitama_acc.amount
+```
+
+```
+saitama_acc.balance()
+```
+
+### Setters
+A setter is a method that sets the value of a property.
+```
+  def deposit(self, amount):
+    self.amount += amount
+
+  def withdraw(self, amount):
+    self.amount -= amount
+```
+
+```
+saitama_acc.deposit(500)
+saitama_acc.withdraw(320)
+```
+
+
+## Inheritance
+Inheritance enables us to create a new class from an existing class. The new class (child class) is a specialized version of the existing class (parent class) and it inherits all the non-private variables and methods of the existing class.
+
+If no additional instance variable(s) are required in the child class, `__init__()` can be left out.
+```
+class SavingsAccount(BankAccount):
+  pass
+```
+
+If additional instance variable(s) are required in the child class, `super()` is used.
```

Comment:
This sentence is a bit misleading. 

super is used in any situation where we would want to make use of functionality defined inside the parent class, within a function with the same name. 

Eg if you want to make a new __init__ function without completely throwing away the parent class __init__ function. Or if you want to make a new withdraw function that makes use of the parent's withdraw function. 

super isn't only for initialisers.